File: src/main/java/leets/weeth/domain/penalty/application/usecase/PenaltyUsecase.java
package leets.weeth.domain.penalty.application.usecase;

import leets.weeth.domain.penalty.application.dto.PenaltyDTO;

import java.util.List;

public interface PenaltyUsecase {

    void save(PenaltyDTO.Save dto);

    void update(PenaltyDTO.Update dto);

    List<PenaltyDTO.ResponseAll> findAll(Integer cardinalNumber);

    PenaltyDTO.Response find(Long userId);

    void delete(Long penaltyId);

}

File: src/main/kotlin/com/weeth/domain/penalty/application/usecase/PenaltyUsecase.kt
package com.weeth.domain.penalty.application.usecase

import com.weeth.domain.penalty.application.dto.PenaltyDTO

interface PenaltyUsecase {
    fun save(dto: PenaltyDTO.Save)

    fun update(dto: PenaltyDTO.Update)

    fun findAll(cardinalNumber: Int?): List<PenaltyDTO.ResponseAll>

    fun find(userId: Long): PenaltyDTO.Response

    fun delete(penaltyId: Long)
}
